am6uno/Algorithm-Run-Time-Complexity
# Added the ability to restrict page access based on role

### Description
- Routes are now restricted by role.
- The Navbar shows the proper menu options for student and teachers.
- Teacher and student routing after login was fixed.
- Altered the keycloak setup to say encounter at the top of the pages.
![image](https://user-images.githubusercontent.com/106398280/234760576-6e6cdc6a-6afa-4d80-9564-df93080a0cf4.png)

### Tests Steps
- Create a teacher user.
You should the correct navigation options in the nav menu.
You should be routed to the classroom page.

- Create a student user and try to access the old problem search page. You should be able to access this page as a student.
You should the correct navigation options in the nav menu.
You should be routed to the problem set page.
